src/time_tracker/projects/db.clj

```diff
@@ -0,0 +1,81 @@
+(ns time-tracker.projects.db
+  (:require [clojure.java.jdbc :as jdbc]
+
+            [time-tracker.db :as db]))
+
+(defn retrieve-project-if-authorized
+  "Retrieves one specific project if authorized."
+  [project-id google-id]
+  (let [query (str "SELECT project.* FROM project "
+                   "INNER JOIN project_permission ON project.id = project_permission.project_id "
+                   "INNER JOIN app_user ON app_user.id = project_permission.app_user_id "
+                   "WHERE app_user.google_id = ? "
+                   "AND ?::permission = ANY (project_permission.permissions) "
+                   "AND project.id = ?;")
+        connection (db/connection)]
+    (first (jdbc/query connection [query google-id "admin" project-id]))))
+
+(defn update-project-if-authorized!
+  "Updates a project if authorized and returns the updated project."
+  [project-id contents google-id]
+  (let [update-query (str "UPDATE project "
+                          "SET name = ? "
+                          "FROM project_permission, app_user "
+                          "WHERE project_permission.project_id = project.id "
+                          "AND project_permission.app_user_id = app_user.id "
+                          "AND ?::permission = ANY (project_permission.permissions) "
+                          "AND app_user.google_id = ? "
+                          "AND project.id = ?;")]
+    ;; The Postgres RETURNING clause doesn't work,
+    ;; so using two queries for now
+    (jdbc/with-db-transaction [connection (db/connection)]
```

Comment:
You might want to consider moving transactions to the middleware level. That way a request-response cycle is a single transaction, and you don't miss out pulling in random reads into a transaction. This is typical of CRUD apps.